src/kbmod/work_unit.py

```diff
@@ -773,9 +710,83 @@ def from_sharded_fits(cls, filename, directory, lazy=False):
             per_image_indices=per_image_indices,
             lazy=lazy,
             file_paths=file_paths,
+            obstimes=obstimes,
         )
         return result
 
+    def metadata_to_primary_header(self, include_wcs=True):
+        """Creates the metadata fits headers.
+
+        Parameters
+        ----------
+        include_wcs : `bool`
+            whether or not to append all the per image wcses
+            to the header (optional for the serial `to_fits`
+            case so that we can maintain the same indexing
+            as before).
+        Returns
+        -------
+        hdul : `astropy.io.fits.HDUList`
+            The HDU List.
+        """
+        # Set up the initial HDU list, including the primary header
+        # the metadata (empty), and the configuration.
+        hdul = fits.HDUList()
+        pri = fits.PrimaryHDU()
+        pri.header["NUMIMG"] = self.get_num_images()
+        pri.header["REPRJCTD"] = self.reprojected
+        pri.header["HELIO"] = self.heliocentric_distance
+        for i in range(len(self.constituent_images)):
+            pri.header[f"GEO_{i}"] = self.geocentric_distances[i]
+
+        # If the global WCS exists, append the corresponding keys.
+        if self.wcs is not None:
+            append_wcs_to_hdu_header(self.wcs, pri.header)
+
+        n_constituents = len(self.constituent_images)
+        pri.header["NCON"] = n_constituents
```

Comment:
Is there a reason this is two lines?

Reply:
Oh, I think I see what's going on here. This was abstracted out of the `to_fits` and `to_sharded_fits` methods.